keps/sig-cli/2906-kustomize-function-catalog/README.md

```diff
@@ -0,0 +1,667 @@
+
+In addition to the new catalog kind, `kustomize build` will accept a repeatable flag `--trusted-plugin-catalog=""`. When present, this flag instructs `kustomize build` to automatically fetch and execute plugins that are defined by the catalog and referenced within the Kustomization, Component or Composition. When a resource is processed by `kustomize build` and a catalog is referenced but not specified using the `--trusted-plugin-catalog=""` flag, an error will occur. Kustomize will provide a built in Catalog for supporting official extensions, published to a well publicized endpoint. This catalog will _NOT_ require the user to explicitly trust it. Users can provide the `apiVersion` and `kind` of the official extensions in kustomize resource and these will be resolved by the official catalog.
+
+In addition to container based plugin providers, the `Catalog` will support discovery of Starlark and Exec based plugin providers, via an HTTP(s), Git, or OCI reference as illustrated below: 
+
+```yaml
+apiVersion: kustomize.io/v1
+kind: Catalog
+metadata: 
+  name: "example-co-plugins"
+spec: 
+  modules: 
+  - apiVersion: example.com/v1
+    kind: GroovyApplication
+    description: "A Kustomize plugin provider that can handle groovy apps"
+    provider:  
+      starlark: https://example.co/module_providers/starlark-func:v1.0.0
+```
+
+This concept can be extended later to support additional plugin provider packaging, but is out of scope for the current proposal.
+
+When HTTP(s) references are used, the HTTP(s) endpoint must support anonymous access for reading resources. Resources will be expected to be stored as a single file, such as `catalog.yaml`. Git or OCI references can be authenticated or anonymous, and will use appropriate configuration from the users file-system. 
+
+ 
+### User Stories (Optional)
+
+#### Story 1
+
+As a platform developer at enterprise company Example Co, I want to publish a catalog of Kustomize plugin providers that represent custom capabilities important for our internal Kubernetes platform. I have built and published several Kustomize plugins, packaged as Docker images, to our internal Docker registry: `docker.example.co`.
+
+To do this, I build a new `Catalog` API resource: 
+
+```yaml
+# catalog.yaml
+apiVersion: kustomize.io/v1
+kind: Catalog
+metadata: 
+  name: "example-co-plugins"
+spec: 
+  modules: 
+  - apiVersion: example.com/v1
+    kind: JavaApplication
+    description: "A Kustomize plugin provider that can handle Java apps"
+    provider: 
+      container: 
+        image: docker.example.co/plugins/java:v1.0.0
```

Comment:
Question, will this support fields such as `network` or `mounts`? I think these are necessary additions to the catalog UX if we want to support containerized functions in a meaningful way. 

For mounts, how would these filepaths be resolved? Would they be relative to the catalog file, or relative to the CR file invoking the function?